Sources/AsyncHTTPClient/ConnectionsState.swift

```diff
@@ -148,7 +148,7 @@ extension HTTP1ConnectionProvider {
                     return .none
                 }
             case .closed:
-                return .fail(waiter, ProviderClosedError())
+                return .fail(waiter, HTTPClientError.alreadyShutdown)
```

Comment:
@artemredkin from what I can tell, no test noticed that this error changed which probably means no test is getting this error ;). I think we should probably add a test case that tests this condition.

Reply:
filed #244

Reply:
Specific test is a good idea, yes, though this error was picked up by a test, just not reliably :)